File: Boare.Lib.Vsq/UstPortamento.cs
/*
 * UstPortamento.cs
 * Copyright (c) 2009 kbinani
 *
 * This file is part of Boare.Lib.Vsq.
 *
 * Boare.Lib.Vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * Boare.Lib.Vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
#if JAVA
package org.kbinani.vsq;

import java.util.*;
import java.io.*;
import org.kbinani.*;
#else
using System;
using bocoree;
using bocoree.util;
using bocoree.io;

namespace Boare.Lib.Vsq {
#endif

#if JAVA
    public class UstPortamento implements Cloneable, Serializable {
#else
    [Serializable]
    public class UstPortamento : ICloneable {
#endif
        public Vector<UstPortamentoPoint> Points = new Vector<UstPortamentoPoint>();
        public int Start;

        public void print( BufferedWriter sw )
#if JAVA
            throws IOException
#endif
        {
            String pbw = "";
            String pby = "";
            String pbm = "";
            int count = Points.size();
            for ( int i = 0; i < count; i++ ) {
                String comma = (i == 0 ? "" : ",");
                pbw += comma + Points.get( i ).Step;
                pby += comma + Points.get( i ).Value;
                String type = "";
                UstPortamentoType ut = Points.get( i ).Type;
                if ( ut == UstPortamentoType.S ) {
                    type = "";
                } else if ( ut == UstPortamentoType.Linear ) {
                    type = "s";
                } else if ( ut == UstPortamentoType.R ) {
                    type = "r";
                } else if ( ut == UstPortamentoType.J ) {
                    type = "j";
                }
                pbm += comma + type;
            }
            sw.write( "PBW=" + pbw );
            sw.newLine();
            sw.write( "PBS=" + Start );
            sw.newLine();
            sw.write( "PBY=" + pby );
            sw.newLine();
            sw.write( "PBM=" + pbm );
            sw.newLine();
        }

        public Object clone() {
            UstPortamento ret = new UstPortamento();
            int count = Points.size();
            for ( int i = 0; i < count; i++ ) {
                ret.Points.add( Points.get( i ) );
            }
            ret.Start = Start;
            return ret;
        }

#if !JAVA
        public object Clone() {
            return clone();
        }
#endif

        /*
        PBW=50,50,46,48,56,50,50,50,50
        PBS=-87
        PBY=-15.9,-20,-31.5,-26.6
        PBM=,s,r,j,s,s,s,s,s
        */
        public void ParseLine( String line ) {
            line = line.ToLower();
            String[] spl = PortUtil.splitString( line, '=' );
            if ( spl.Length == 0 ) {
                return;
            }
            String[] values = PortUtil.splitString( spl[1], ',' );
            if ( line.StartsWith( "pbs=" ) ) {
                Start = PortUtil.parseInt( values[0] );
            } else if ( line.StartsWith( "pbw=" ) ) {
                for ( int i = 0; i < values.Length; i++ ) {
                    if ( i >= Points.size() ) {
                        Points.add( new UstPortamentoPoint() );
                    }
                    UstPortamentoPoint up = Points.get( i );
                    up.Step = PortUtil.parseInt( values[i] );
                    Points.set( i, up );
                }
            } else if ( line.StartsWith( "pby=" ) ) {
                for ( int i = 0; i < values.Length; i++ ) {
                    if ( i >= Points.size() ) {
                        Points.add( new UstPortamentoPoint() );
                    }
                    UstPortamentoPoint up = Points.get( i );
                    up.Value = PortUtil.parseFloat( values[i] );
                    Points.set( i, up );
                }
            } else if ( line.StartsWith( "pbm=" ) ) {
                for ( int i = 0; i < values.Length; i++ ) {
                    if ( i >= Points.size() ) {
                        Points.add( new UstPortamentoPoint() );
                    }
                    UstPortamentoPoint up = Points.get( i );
                    String search = values[i].ToLower();
                    if ( search.Equals( "s" ) ) {
                        up.Type = UstPortamentoType.Linear;
                    } else if ( search.Equals( "r" ) ) {
                        up.Type = UstPortamentoType.R;
                    } else if ( search.Equals( "j" ) ) {
                        up.Type = UstPortamentoType.J;
                    } else {
                        up.Type = UstPortamentoType.S;
                    }
                    Points.set( i, up );
                }
            } else if ( line.StartsWith( "pbs=" ) ) {

            }
        }
    }

#if !JAVA
}
#endif

File: port_java_OBSOLUTE/com/boare/vsq/UstPortamento.java
/*
 * UstPortamento.java
 * Copyright (c) 2009 kbinani
 *
 * This file is part of com.boare.vsq.
 *
 * com.boare.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * com.boare.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
package com.boare.vsq;

import java.util.*;
import java.io.*;
import com.boare.corlib.*;

public class UstPortamento implements Cloneable {
    public Vector<UstPortamentoPoint> points = new Vector<UstPortamentoPoint>();
    public int start;

    public static String getXmlElementName( String name ){
        if( name.equals( "points" ) ){
            return "Points";
        }else if( name.equals( "start" ) ){
            return "Start";
        }
        return name;
    }

    public static String getGenericTypeName( String name ){
        if( name.equals( "points" ) ){
            return "com.boare.vsq.UstPortamentoPoint";
        }
        return "";
    }

    public void print( StreamWriter sw ) throws IOException{
        String pbw = "";
        String pby = "";
        String pbm = "";
        int c = points.size();
        for ( int i = 0; i < c; i++ ) {
            UstPortamentoPoint p = points.get( i );
            String comma = (i == 0 ? "" : ",");
            pbw += comma + p.step;
            pby += comma + p.value;
            String type = "";
            switch ( p.type ) {
                case S:
                    type = "";
                    break;
                case Linear:
                    type = "s";
                    break;
                case R:
                    type = "r";
                    break;
                case J:
                    type = "j";
                    break;
            }
            pbm += comma + type;
        }
        sw.writeLine( "PBW=" + pbw );
        sw.writeLine( "PBS=" + start );
        sw.writeLine( "PBY=" + pby );
        sw.writeLine( "PBM=" + pbm );
    }

    public Object clone() {
        UstPortamento ret = new UstPortamento();
        int c = points.size();
        for ( int i = 0; i < c; i++ ) {
            ret.points.add( points.get( i ) );
        }
        ret.start = start;
        return ret;
    }

    /*
    PBW=50,50,46,48,56,50,50,50,50
    PBS=-87
    PBY=-15.9,-20,-31.5,-26.6
    PBM=,s,r,j,s,s,s,s,s
    */
    public void parseLine( String line ) {
        line = line.toLowerCase();
        String[] spl = line.split( "=" );
        if ( spl.length == 0 ) {
            return;
        }
        String[] values = spl[1].split( "," );
        if ( line.startsWith( "pbs=" ) ) {
            start = Integer.parseInt( values[0] );
        } else if ( line.startsWith( "pbw=" ) ) {
            for ( int i = 0; i < values.length; i++ ) {
                if ( i >= points.size() ) {
                    points.add( new UstPortamentoPoint() );
                }
                points.get( i ).step = Integer.parseInt( values[i] );
            }
        } else if ( line.startsWith( "pby=" ) ) {
            for ( int i = 0; i < values.length; i++ ) {
                if ( i >= points.size() ) {
                    points.add( new UstPortamentoPoint() );
                }
                points.get( i ).value = Float.parseFloat( values[i] );
            }
        } else if ( line.startsWith( "pbm=" ) ) {
            for ( int i = 0; i < values.length; i++ ) {
                if ( i >= points.size() ) {
                    points.add( new UstPortamentoPoint() );
                }
                String s = values[i].toLowerCase();
                if( s.equals( "s" ) ){
                    points.get( i ).type = UstPortamentoType.Linear;
                }else if( s.equals( "r" ) ){
                    points.get( i ).type = UstPortamentoType.R;
                }else if( s.equals( "j" ) ){
                    points.get( i ).type = UstPortamentoType.J;
                }else{
                    points.get( i ).type = UstPortamentoType.S;
                }
            }
        } else if ( line.startsWith( "pbs=" ) ) {

        }
    }
}
